### ISSUE TITLE
Purpose of ERROR Details|Dict is unclear

### ISSUE THREAD

**Issue by ecorm** [tavendo/WAMP#120]
It is unclear what the purpose of the `Details|Dict` field is for `ERROR` messages. I've tried using it to send additional error information to the callee, but Crossbar did not propagate that information. I suspect that this field is really intended for future options.

**Comment by oberstet** [comment]
Exactly. With WAMP1, we've learned that extensibility points are important. So yes, this is for future use. This is discussed in the spec, but only "in general". We should probably add text for `ERROR` also ...

**Comment by ecorm** [comment]
Thanks. Under [Invocation ERROR](https://github.com/tavendo/WAMP/blob/master/spec/basic.md#invocation-error) and [Call ERROR](https://github.com/tavendo/WAMP/blob/master/spec/basic.md#call-error) it says:

> `Details` is a dictionary with additional error details.

which is what lead to my confusion. Perhaps those sentences should be revised.

**Comment by gammazero** [comment]
**Purpose of ERROR Details**

`ERROR Details|Dict` provides a mechanism for optionally reporting structured error information through fields instead of long, unparseable error messages.   The `Details|Dict` contains field names and values and is suitable as input for structured logging implementations.
